kubert/src/lease.rs

```diff
@@ -300,7 +307,32 @@ impl LeaseManager {
                 }
 
                 // Update the claim and broadcast it to all receivers.
-                claim = self.ensure_claimed(&claimant, &params).await?;
+                let backoff = new_backoff.with_max_interval(grace).build();
+                claim = backoff::future::retry(backoff, || {
+                    self.ensure_claimed(&claimant, &params).map_err(|err| match err {
+                        err @ Error::Api(kube_client::Error::Auth(_))
+                        | err @ Error::Api(kube_client::Error::Discovery(_))
+                        | err @ Error::Api(kube_client::Error::BuildRequest(_)) => {
+                            backoff::Error::Permanent(err)
+                        },
+                        err @ Error::Api(kube_client::Error::InferConfig(_)) => {
+                            debug_assert!(false, "InferConfig errors should only be returned when constructing a new client");
+                            backoff::Error::Permanent(err)
+                        },
+                        // Retry any other API request errors.
+                        err @ Error::Api(_) => {
+                            tracing::debug!(error = %err, "Error claiming lease, retrying...");
+                            backoff::Error::Transient {
+                                err,
+                                // Allow the backoff implementation to select how
+                                // long to wait before retrying.
+                                retry_after: None,
+                            }
+                        }
+                        err @ Error::MissingResourceVersion | err @ Error::MissingSpec => backoff::Error::Permanent(err),
```

Comment:
I wasn't entirely sure whether these errors should be fatal or transient --- do they imply that the lease resource is fundamentally broken, or that we just got a malformed response once?